csharp/src/EnumerableExt.cs

```diff
@@ -0,0 +1,107 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using JetBrains.Annotations;
+
+namespace ArmoniK.Utils;
+
+public static class EnumerableExt
+{
+  /// <summary>
+  ///   Convert an enumerable into a list, if it is not already a list
+  ///   Beware that the return list may or may not be a reference to the input enumerable
+  /// </summary>
+  /// <param name="enumerable">The enumerable to convert into a list</param>
+  /// <typeparam name="T">Type of the elements</typeparam>
+  /// <returns>A list containing the same elements as the input enumerable</returns>
+  [PublicAPI]
+  public static IList<T> AsList<T>(this IEnumerable<T>? enumerable)
+  {
+    if (enumerable is null)
+    {
+      return Array.Empty<T>();
+    }
+
+    return enumerable as IList<T> ?? enumerable.ToList();
+  }
+
+  /// <summary>
+  ///   Convert an enumerable into a collection, if it is not already a collection
+  ///   Beware that the return collection may or may not be a reference to the input enumerable
+  /// </summary>
+  /// <param name="enumerable">The enumerable to convert into a collection</param>
+  /// <typeparam name="T">Type of the elements</typeparam>
+  /// <returns>A collection containing the same elements as the input enumerable</returns>
+  [PublicAPI]
+  public static ICollection<T> AsCollection<T>(this IEnumerable<T>? enumerable)
+  {
+    if (enumerable is null)
+    {
+      return Array.Empty<T>();
+    }
+
+    return enumerable as ICollection<T> ?? enumerable.ToList();
+  }
+
+  /// <summary>
+  ///   Split the elements of a sequence into chunks of size at most <paramref name="size" />.
+  /// </summary>
+  /// <remarks>
+  ///   Every chunk except the last will be of size <paramref name="size" />.
+  ///   The last chunk will contain the remaining elements and may be of a smaller size.
+  /// </remarks>
+  /// <param name="source">
+  ///   An <see cref="IEnumerable{T}" /> whose elements to chunk.
+  /// </param>
+  /// <param name="size">
+  ///   Maximum size of each chunk.
+  /// </param>
+  /// <typeparam name="TSource">
+  ///   The type of the elements of source.
+  /// </typeparam>
+  /// <returns>
+  ///   An <see cref="IEnumerable{T}" /> that contains the elements the input sequence split into chunks of size
+  ///   <paramref name="size" />.
+  /// </returns>
+  /// <exception cref="ArgumentOutOfRangeException">
+  ///   <paramref name="size" /> is below 1.
+  /// </exception>
+  [PublicAPI]
+  public static IEnumerable<TSource[]> AsChunked<TSource>(this IEnumerable<TSource>? source,
```

Comment:
In .Net, usually we prefer use a verb for methods/Functions and a name for classes.
The AsXXXX and ToXXXXX Patterns are exceptions.
Hence here, renamt to `Chunk()` would be more natural.

Reply:
`Chunk` name is in conflict with the the function from LinQ, which we don't know if it's there or not. In Extension.Csharp, we named it `ToChunk` I think. I am not sure what is the best name.

Reply:
I'd still use `Chunk` and let the `using` in client code make the choice.

Reply:
I just checked, and it is not possible to disambiguate with `using`. Renamed as `ToChunks`.